File: Funs/transform.m
function Matrix_T = transform(Matrix,type)

switch type
    case 'value2trans'
        [~, sortedIndices] = sort(Matrix, 2);

        OrdinalNumbers = repmat(1:size(Matrix, 2), size(Matrix, 1), 1);

        linearIndex = sub2ind(size(Matrix), repmat((1:size(Matrix, 1))', 1, size(Matrix, 2)), sortedIndices);

        Matrix_T = zeros(size(Matrix)); Matrix_T(linearIndex) = (OrdinalNumbers/size(Matrix, 2)).^0.5;

    case 'cosine'
        Matrix_T = Matrix./sqrt(sum(Matrix.^2,2));

    case 'log'
        Matrix_T = log2(Matrix + 1);

end

File: Python/transform.py
import numpy as np

def transform(Matrix, type_):
    if type_ == 'value2trans':
        sorted_indices = np.argsort(Matrix, axis=1, kind='stable')
        ordinal_numbers = np.tile(np.arange(1, Matrix.shape[1] + 1), (Matrix.shape[0], 1))
        linear_index = np.arange(Matrix.shape[0])[:, None] * Matrix.shape[1] + sorted_indices
        Matrix_T = np.zeros_like(Matrix)
        Matrix_T.flat[linear_index] = (ordinal_numbers / Matrix.shape[1]) ** 0.5

    elif type_ == 'cosine':
        Matrix_T = Matrix / np.sqrt(np.sum(Matrix ** 2, axis=1, keepdims=True))

    elif type_ == 'log':
        Matrix_T = np.log2(Matrix + 1)

    return Matrix_T